### ISSUE TITLE
V0.25.0rc3 Task details contains a null value when it can contains a finite value

### ISSUE THREAD

**Issue by gmourier** [meilisearch/MeiliSearch#2053]
**Describe the bug**

The value of `indexedDocuments` could be `0` rather than `null` when a task fails.

```json
{
            "uid": 1,
            "indexUid": "test",
            "status": "failed",
            "type": "documentAddition",
            "details": {
                "receivedDocuments": 2500,
                "indexedDocuments": null
            },
            "error": {
                "message": "Index `test` not found.",
                "code": "index_not_found",
                "type": "invalid_request",
                "link": "https://docs.meilisearch.com/errors#index_not_found"
            },
            "duration": "PT0.000578S",
            "enqueuedAt": "2022-01-04T12:09:36.709163Z",
            "startedAt": "2022-01-04T12:09:36.709532Z",
            "finishedAt": "2022-01-04T12:09:36.709741Z"
}
```

**To Reproduce**
1. Add documents on an inexistent index with an API key that do not have the right to create an index. (It also happens if the user tries to push documents missing a document identifier so it's probably not related to that particular case)
2. Fetch the related task

**Expected behavior**

When a real value can be given, it should be preferred over `null` in the details of a task for the field `indexedDocuments` and `deletedDocuments`. In this particular case, we know that 0 documents have been indexed.

**MeiliSearch version:** [e.g. v0.25.0rc3]

**Comment by curquiza** [comment]
Thanks @gmourier, I assign the Capitaine Haddock on it :)

**Comment by curquiza** [comment]
Closed by #2059
